OpenEdXMobile/res/layout/fragment_course_unit_video.xml

```diff
@@ -63,12 +63,19 @@
         android:paddingBottom="@dimen/widget_margin_half"
         android:paddingTop="@dimen/widget_margin_half" />
 
-    <fragment
-        android:id="@+id/cast_mini_controller"
-        class="com.google.android.gms.cast.framework.media.widget.MiniControllerFragment"
+    <FrameLayout
+        android:id="@+id/fl_mini_controller"
         android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:visibility="gone" />
+        android:visibility="gone"
+        android:layout_height="wrap_content">
+
+        <fragment
+            android:id="@+id/cast_mini_controller"
+            class="com.google.android.gms.cast.framework.media.widget.MiniControllerFragment"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:visibility="gone" />
+    </FrameLayout>
```

Comment:
Cast Mini controller visibility is controlled by the SDK, and in case of YouTube in-app player screen app needs to play the video within the app, either Connected with the cast device or not. 

So we need to hide the mini controller when another video is casting on the remote device.

Reply:
Application is crashing while placing the `**MiniControllerFragment**` in the container in Runtime, and there is no other way mention in the Chrome cast doc.